src/shared/Microsoft.Git.CredentialManager/Authentication/MicrosoftAuthentication.cs

```diff
@@ -10,15 +10,22 @@ namespace Microsoft.Git.CredentialManager.Authentication
 {
     public interface IMicrosoftAuthentication
     {
-        Task<string> GetAccessTokenAsync(string authority, string clientId, Uri redirectUri, string resource);
+        Task<string> GetAccessTokenAsync(string authority, string clientId, Uri redirectUri, string resource, Uri remoteUri);
     }
 
     public class MicrosoftAuthentication : AuthenticationBase, IMicrosoftAuthentication
     {
+        public static readonly string[] AuthorityIds =
+        {
+            "MSA",  "Microsoft",   "MicrosoftAccount",
+            "AAD",  "Azure",       "AzureDirectory",
+            "Live", "LiveConnect", "LiveID",
+        };
```

Comment:
Same question as the other comment, where did you get this list of AuthorityIds from?

Reply:
From the Windows GCM.. these are all the supported values :(

Not sure why we didn't just allow one option per authority, since they all end up returning the same thing!

https://github.com/microsoft/Git-Credential-Manager-for-Windows/blob/9270c89a0ce109c7a86e9b3d8174a0af56aab468/Shared/Cli/Functions/Common.cs#L433-L474

Reply:
Can/should we simplify?

Reply:
@jrbriggs
We have no idea which values are being used actively to know which ones to drop.

I we want to _really_ simplify we can just break back-compat and ignore these values. Users would need to manually migrate to specifying the provider.